databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -0,0 +1,484 @@
+# Databricks CLI
+# Copyright 2021 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"), except
+# that the use of services to which certain application programming
+# interfaces (each, an "API") connect requires that the user first obtain
+# a license for the use of the APIs from Databricks, Inc. ("Databricks"),
+# by creating an account at www.databricks.com and agreeing to either (a)
+# the Community Edition Terms of Service, (b) the Databricks Terms of
+# Service, or (c) another written agreement between Licensee and Databricks
+# for the use of the APIs.
+#
+# You may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import click
+
+from databricks_cli.click_types import JsonClickType
+from databricks_cli.configure.config import provide_api_client, profile_option, debug_option
+from databricks_cli.unity_catalog.api import UnityCatalogApi
+from databricks_cli.unity_catalog.utils import mc_pretty_format
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, json_cli_base
+
+
+##############  Share Commands  ##############
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
```

Comment:
nit: probably not for the scope of this PR, but for the future improvements: we should create one single decorator, that generalises the use of all of these options and parses docstring for help text.

Reply:
+1 -- lots of repetition here and in the rest of the CLI for these annotations.

Reply:
Filed #472.